### PR TITLE
Add linux-deb & linux-rpm build artifacts to daily-build & release workflows

### PR DESCRIPTION
**Pull request by NipunaMadhushan**
## Purpose
Create linux-deb & linux-rpm build artifacts in the release workflow

## Goals
Remove jenkins from the release pipeline.

### CODE REVIEW

File: installers/mac/build-ballerina-macos-x64.sh [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    rm -rf target
+}
+
+function extractPack() {
+    echo "Extracting the ballerina distribution, " $1
+    rm -rf target/original
+    mkdir -p target/original
+    unzip $1 -d target/original > /dev/null 2>&1
+
+    cert="Developer ID Installer: WSO2, Inc. (QH8DVR4443)"
```
Comment (keizer619) [inline installers/mac/build-ballerina-macos-x64.sh:82]:
```suggestion
```

File: installers/mac/build-ballerina-macos-x64.sh [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        mkdir temp
+        cd temp
+        file_name=$(echo "$jarFile" | sed "s/.*\///")
+        cp ../$jarFile .
+        unzip -o $file_name
+
+        code=1
+        for lib in $(find ../temp -name '*.jnilib' -o -name '*.dylib'); do
+            echo "Signing $lib"
+            codesign -s "$cert" --deep --options=runtime --force $lib
```
Comment (keizer619) [inline installers/mac/build-ballerina-macos-x64.sh:93]:
```suggestion
```

File: installers/mac/build-ballerina-macos-x64.sh [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                echo "$lib has signed"
+            else
+                echo "Signing $lib has failed with exit code $code"
+            fi 
+        done
+
+        if [ $code -eq 0 ]; then
+            rm $file_name
+            zip -r $file_name *
+            codesign -s "$cert" --deep --options=runtime --force $file_name
```
Comment (keizer619) [inline installers/mac/build-ballerina-macos-x64.sh:105]:
```suggestion
```

File: installers/linux-deb/build-ballerina-linux-deb-x64.sh [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+# function createBallerinaRuntime() {
+#     echo "Creating ballerina runtime installer"
+
+#     extractPack "$BALLERINA_DISTRIBUTION_LOCATION/$BALLERINA_RUNTIME.zip" ${BALLERINA_RUNTIME}
+#     createPackInstallationDirectory
+#     copyDebianDirectory
+#     mv target/${BALLERINA_INSTALL_DIRECTORY} target/ballerina-runtime-linux-installer-x64-${BALLERINA_VERSION}
+#     fakeroot dpkg-deb --build target/ballerina-runtime-linux-installer-x64-${BALLERINA_VERSION}
+# }
```
Comment (keizer619) [inline installers/linux-deb/build-ballerina-linux-deb-x64.sh:127]:
```suggestion
```

File: installers/mac/build-ballerina-macos-x64.sh [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+    for jarFile in $(find target/original/ballerina-macos-$BALLERINA_VERSION -name '*.jar'); do
+        mkdir temp
+        cd temp
+        file_name=$(echo "$jarFile" | sed "s/.*\///")
+        cp ../$jarFile .
+        unzip -o $file_name
+
+        code=1
+        for lib in $(find ../temp -name '*.jnilib' -o -name '*.dylib'); do
```
Comment (NipunaMadhushan) [inline installers/mac/build-ballerina-macos-x64.sh:90]: @keizer619 Should this for loop be removed?

Comment (keizer619) [reply installers/mac/build-ballerina-macos-x64.sh:90]: Yes lets remove this as well

File: installers/linux-rpm/build-ballerina-linux-rpm-x64.sh [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    echo "Creating all distributions"
+    createBallerinaPlatform
+    # createBallerinaRuntime
+else
+    if [ "$DISTRIBUTION" == "ballerina" ]; then
+        echo "Creating Ballerina Platform"
+        createBallerinaPlatform
+    # elif [ "$DISTRIBUTION" == "ballerina-runtime" ]; then
+    #     echo "Creating Ballerina Runtime"
+    #     createBallerinaRuntime
```
Comment (keizer619) [inline installers/linux-rpm/build-ballerina-linux-rpm-x64.sh:125]:
```suggestion
```

File: installers/linux-rpm/build-ballerina-linux-rpm-x64.sh [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    [ -f ${PLATFORM_SPEC_FILE_LOC} ] && rm -f ${PLATFORM_SPEC_FILE_LOC}
+    cp resources/${PLATFORM_SPEC_FILE} ${SPEC_FILES_LOCATION}
+    setupVersion_platform
+    rpmbuild -bb --define "_topdir  $(pwd)/rpmbuild" ${PLATFORM_SPEC_FILE_LOC}
+}
+
+if [ "$BUILD_ALL_DISTRIBUTIONS" == "true" ]; then
+    echo "Creating all distributions"
+    createBallerinaPlatform
+    # createBallerinaRuntime
```
Comment (keizer619) [inline installers/linux-rpm/build-ballerina-linux-rpm-x64.sh:118]:
```suggestion
```

File: installers/linux-deb/build-ballerina-linux-deb-x64.sh [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    echo "Creating all distributions"
+    createBallerinaPlatform
+    # createBallerinaRuntime 
+else
+    if [ "$DISTRIBUTION" == "ballerina" ]; then
+        echo "Creating Ballerina Platform"
+        createBallerinaPlatform
+    # elif [ "$DISTRIBUTION" == "ballerina-runtime" ]; then
+    #     echo "Creating Ballerina Runtime"
+    #     createBallerinaRuntime
```
Comment (keizer619) [inline installers/linux-deb/build-ballerina-linux-deb-x64.sh:131]:
```suggestion
```

**Review by keizer619 — APPROVED** [review]

**Pull request merged**